plugin/pkg/managedseed/validator/admission.go

```diff
@@ -239,13 +240,83 @@ func (v *ManagedSeed) Admit(ctx context.Context, a admission.Attributes, o admis
 		allErrs = append(allErrs, errs...)
 	}
 
+	switch a.GetOperation() {
+	case admission.Create:
+		errs, err := v.validateManagedSeedCreate(managedSeed, shoot)
+		if err != nil {
+			return err
+		}
+		allErrs = append(allErrs, errs...)
+	case admission.Update:
+		oldManagedSeed, ok := a.GetOldObject().(*seedmanagement.ManagedSeed)
+		if !ok {
+			return apierrors.NewInternalError(errors.New("could not covert old resource into ManagedSeed object"))
+		}
+		errs, err := v.validateManagedSeedUpdate(oldManagedSeed, managedSeed, shoot)
+		if err != nil {
+			return err
+		}
+		allErrs = append(allErrs, errs...)
+	}
+
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(gk, managedSeed.Name, allErrs)
 	}
 
 	return nil
 }
 
+func (v *ManagedSeed) validateManagedSeedCreate(managedSeed *seedmanagement.ManagedSeed, shoot *gardencore.Shoot) (field.ErrorList, error) {
+	allErrs := field.ErrorList{}
+
+	seedSpec, err := seedmanagementhelper.ExtractSeedSpec(managedSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	shootZones := helper.GetAllZonesFromShoot(shoot)
+
+	if len(seedSpec.Provider.Zones) > 0 && !sets.New[string](seedSpec.Provider.Zones...).Equal(shootZones) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "gardenlet", "config", "seedConfig", "spec", "provider", "zones"), seedSpec.Provider.Zones, fmt.Sprintf("seed provider zones must be equal to shoot zones (%v)", sets.List(shootZones))))
+	}
+
+	return allErrs, nil
+}
+
+func (v *ManagedSeed) validateManagedSeedUpdate(oldManagedSeed, newManagedSeed *seedmanagement.ManagedSeed, shoot *gardencore.Shoot) (field.ErrorList, error) {
+	allErrs := field.ErrorList{}
+	zonesFieldPath := field.NewPath("spec", "gardenlet", "config", "seedConfig", "spec", "provider", "zones")
+
+	oldSeedSpec, err := seedmanagementhelper.ExtractSeedSpec(oldManagedSeed)
+	if err != nil {
+		return nil, err
+	}
+	newSeedSpec, err := seedmanagementhelper.ExtractSeedSpec(newManagedSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := admissionutils.ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, newManagedSeed.Name, v.shootLister, "ManagedSeed"); err != nil {
+		allErrs = append(allErrs, field.Forbidden(zonesFieldPath, "zones must not be removed while shoots are still scheduled onto seed"))
+	}
```

Comment:
Don't we need to add validation to the `ManagedSeedShoot` admission plugin that you are not allowed to remove a zone from ManagedSeed's Shoot when this zone exists in the ManagedSeed's spec?

Reply:
From my experience, usually validation added to the `ManagedSeed` admission plugin that includes the Shoot has to be added to the `ManagedSeedShoot` admission plugin as well to make sure that you are not able to update the Shoot spec with the wrong configuration.

Reply:
Nice catch! Added in [7d681be](https://github.com/gardener/gardener/pull/7695/commits/7d681be5d6c1a792c929238e611285df14d0e982).